Bugsnag/Client/BugsnagClient.m

```diff
@@ -163,10 +162,17 @@ void BSSerializeDataCrashHandler(const BSG_KSCrashReportWriter *writer, int type
         if (bsg_g_bugsnag_data.stateJSON) {
             writer->addJSONElement(writer, "state", bsg_g_bugsnag_data.stateJSON);
         }
-        if (bsg_g_bugsnag_data.breadcrumbsPath) {
-            // FIXME: This needs to be updated to cater for new breadcrumb storage scheme
-            // FIXME: addJSONFileElement generates broken JSON if the file cannot be opened
-            // writer->addJSONFileElement(writer, "breadcrumbs", bsg_g_bugsnag_data.breadcrumbsPath);
+        if (bsg_g_bugsnag_data.breadcrumbs) {
+            writer->beginArray(writer, "breadcrumbs");
+            BugsnagBreadcrumbsContext *breadcrumbs = bsg_g_bugsnag_data.breadcrumbs;
+            for (int i = breadcrumbs->firstFileNumber; i < breadcrumbs->nextFileNumber; i++) {
+                static char path[PATH_MAX];
```

Comment:
`static` would make this not thread-safe

Reply:
Will allocate it on the stack 👍